Tests/GRPCCodeGenTests/Internal/Renderer/TextBasedRendererTests.swift

```diff
@@ -356,6 +372,16 @@ final class Test_TextBasedRenderer: XCTestCase {
     )
   }
 
+  func testEmptyLine() throws {
+    try _test(
+      .emptyLine,
+      renderedBy: TextBasedRenderer.renderCodeBlockItem(_:),
+      rendersAs: #"""
+
+        """#
```

Comment:
I don't think the test tests what you think it tests 🙂

The last line of a multiline string literal _doesn't_ include a line break. As the line is otherwise empty it means that the multiline string in your test is really just an empty string. (You should be able to replace it with `""` and the test would be the same.)

---

From https://docs.swift.org/swift-book/documentation/the-swift-programming-language/stringsandcharacters#Multiline-String-Literals

> A multiline string literal includes all of the lines between its opening and closing quotation marks. The string begins on the first line after the opening quotation marks (""") and ends on the line before the closing quotation marks, which means that neither of the strings below start or end with a line break:
> ```swift
> let singleLineString = "These are the same."
> let multilineString = """
> These are the same.
> """
> ```

Reply:
FWIW the correct test here should be rendering as `"\n"`